### PR TITLE
Update staging of Python wheels

### PR DESCRIPTION
**Pull request by AnandInguva**
Fixes: https://github.com/apache/beam/issues/24110

Apache beam releases manylinux_2014 wheels from `2.43.0` release. Update the tag on the stager to match the wheel pattern.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/runners/portability/stager.py [hunk 1]
```diff
@@ -897,10 +901,10 @@ def _create_beam_sdk(sdk_remote_location, temp_dir):
   def _download_pypi_sdk_package(
       temp_dir,
       fetch_binary=False,
-      language_version_tag='27',
+      language_version_tag='39',
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/portability/stager.py:904]: doesn't matter here, but in general changing default values may be a risky change in case there are some callers that rely on older default.

File: sdks/python/apache_beam/runners/portability/stager.py [hunk 2]
```diff
@@ -935,7 +939,12 @@ def _download_pypi_sdk_package(
           '--platform',
           platform_tag
       ])
-      # Example wheel: apache_beam-2.4.0-cp27-cp27mu-manylinux1_x86_64.whl
+      # Example wheel: apache_beam-2.42.0-cp39-cp39-manylinux1_x86_64.whl
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/portability/stager.py:942]: no longer need to mention the outdated case?

**Comment by AnandInguva** [comment]
Dataflow runner able to catch the cached wheel wit these changes. Without these changes, there is a 5 minutes overhead for installing Apache beam for sdist. 

https://screenshot.googleplex.com/uvouKrEiLFD2eEJ

File: sdks/python/apache_beam/runners/portability/stager.py [hunk 3]
```diff
@@ -935,7 +939,12 @@ def _download_pypi_sdk_package(
...
+      # Example wheel: with manylinux14 tag.
+      # apache_beam-2.43.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/portability/stager.py:944]: nit: you can move it on the same line for readability and disable linter on this line if necessary

**Review by tvalentyn — APPROVED** [review]
could you add some details to the PR description what pypi changes required us making these modifications?

Also to confirm, we don't need any changes to staging requirements cache, correct?

**Comment by tvalentyn** [comment]
> could you add some details to the PR description what pypi changes required us making these modifications?

looks like you updated it already.do you know what changes made us build manylinux2014 wheels?

**Comment by AnandInguva** [comment]
> description

No, we don't need to change anything related to staging requirements cache.

**Comment by AnandInguva** [comment]
> > could you add some details to the PR description what pypi changes required us making these modifications?
> 
> looks like you updated it already.do you know what changes made us build manylinux2014 wheels?

Hm..I think I found why. We updated the `cibuildwheels` to `2.9.0` recently. Before that, it was `1.1.x`. After the upgrade, one thing we missed is how `cibuildwheel` new version builds the wheel. In `cibuildwheel==2.3.0`, they made the default of building images on `manylinux2014` instead of `manylinux2010`

File: sdks/python/apache_beam/runners/portability/stager.py (hunk 1 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/runners/portability/stager.py:904]: Ack.

File: sdks/python/setup.py [hunk 4]
```diff
@@ -95,7 +95,7 @@ def get_version():
 different technologies and user communities.
 '''
 
-RECOMMENDED_MIN_PIP_VERSION = '7.0.0'
```
Comment (chamikaramj) [inline sdks/python/setup.py:98]: Is this change needed for 2.43.0 ? I wonder if this can result in other regressions.

**Comment by chamikaramj** [comment]
Could you please come up with a minimum change that can unblock 2.43.0 ? (for example, just change wheel selection to pick up the correct one in stager).

If this is the minimal change that's OK.

**Comment by chamikaramj** [comment]
Thanks. Let's get the change into master first and cherry-pick from there.

**Comment by AnandInguva** [comment]
> Could you please come up with a minimum change that can unblock 2.43.0 ? (for example, just change wheel selection to pick up the correct one in stager).
> 
> If this is the minimal change that's OK.

Unfortunately, this is the minimum change. 

I had to update the regex to accommodate Python 3.10 wheels as well

**Comment by chamikaramj** [comment]
Run Python 3.10 PostCommit

**Comment by chamikaramj** [comment]
Run Python 3.8 PostCommit

**Comment by chamikaramj** [comment]
Run Python Dataflow ValidatesRunner

**Comment by chamikaramj** [comment]
Run Python Dataflow V2 ValidatesRunner

File: sdks/python/setup.py (hunk 4 above)
Comment (AnandInguva) [reply sdks/python/setup.py:98]: Yes, pip can discover manylinux_2014 wheels from version 19.3.

Comment (tvalentyn) [reply sdks/python/setup.py:98]: it's harmless, used below in a warning message

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
this test should not affect postcommit / vr suite since the said suite use beam from sources

**Comment by tvalentyn** [comment]
happy to merge when precommits pass

**Comment by chamikaramj** [comment]
Ack. Thanks.
